File: service-backbase-java/src/main/java/org/example/backend/model/request/query_params/ArticlesListQueryParams.java
package org.example.backend.model.request.query_params;

import lombok.Setter;
import lombok.experimental.Accessors;
import pl.net.testit.serum.api.request.RequestQueryParams;

@Accessors(chain = true)
@Setter
public class ArticlesListQueryParams extends RequestQueryParams {

  public String tag;
  public String author;
  public String favorited;
  public Integer limit;
  public Integer offset;

}

File: service-backbase-kotlin/src/main/kotlin/org/example/backend/model/request/query_params/ArticlesListQueryParams.kt
package org.example.backend.model.request.query_params

import pl.net.testit.serum.api.request.RequestQueryParams


class ArticlesListQueryParams(
        val tag: String? = null,
        val author: String? = null,
        val favorited: String? = null,
        val limit: Int? = null,
        val offset: Int? = null,
) : RequestQueryParams()